arangod/RocksDBEngine/RocksDBIndexCacheRefillFeature.cpp

```diff
@@ -344,9 +349,19 @@ void RocksDBIndexCacheRefillFeature::waitForCatchup() {
 }
 
 void RocksDBIndexCacheRefillFeature::stopThreads() {
+  for (auto& t : _backgroundThreads) {
+    t->beginShutdown();
+    while (t->isRunning()) {
+      std::this_thread::yield();
+    }
```

Comment:
You can just call `shutdown()` - this will block until the thread has terminated.